### ISSUE TITLE
Failed to EstimateGas by `both gasPrice and (maxFeePerGas or maxPriorityFeePerGas) specified`

### ISSUE THREAD

**Issue by yoshidan** [datachainlab/ethereum-ibc-relay-chain#41]
I had a `failed to estimate gas` when I called CreateClient.
The cause is `both gasPrice and (maxFeePerGas or maxPriorityFeePerGas) specified` here is the log.
```json
{"time":"2024-04-25T20:00:12.83226+09:00","level":"ERROR","source":{"function":"github.com/hyperledger-labs/yui-relayer/log.(*RelayLogger).Error","file":"/Users/yoshidan/go/pkg/mod/github.com/hyperledger-labs/yui-relayer@v0.4.22/log/slog.go","line":72},"msg":"failed to estimate gas","module":"ethereum.chain","chain_id":"ibc0","error":"both gasPrice and (maxFeePerGas or maxPriorityFeePerGas) specified","stack":"both gasPrice and (maxFeePerGas or maxPriorityFeePerGas) specified\n(1) attached stack trace\
```

https://github.com/datachainlab/ethereum-ibc-relay-chain/blob/7a848629d64e41e1faa6beef3ad9cc6f7716354c/pkg/relay/ethereum/tx.go#L50

In ethereum, this is thought to occur here.
https://github.com/ethereum/go-ethereum/blob/b20b4a71598481443d60b261d3e5dcb37f8a0d82/internal/ethapi/transaction_args.go#L137

The reason is that `gasPrice` and `gasTipCap` and `gasFeeCap` of the [returned `tx` from `SendTx`](https://github.com/datachainlab/ethereum-ibc-relay-chain/blob/7a848629d64e41e1faa6beef3ad9cc6f7716354c/pkg/relay/ethereum/tx.go#L45)  is not null.

<img width="672" alt="スクリーンショット 2024-04-25 20 15 02" src="https://github.com/datachainlab/ethereum-ibc-relay-chain/assets/958174/5c1598ee-f02f-4773-82be-3c14b157d1be">

### Chain
* bsc v1.2.10 and v1.4.5

### Module
* go-ethereum v1.13.8
* ethereum-ibc-relaye-chain v0.2.14 and [latest](https://github.com/siburu/ethereum-ibc-relay-chain/tree/ibc-go-v8_2)

**Comment by bluele** [comment]
The same problem seems to occur using besu. @siburu

**Comment by siburu** [comment]
Thank you for reporting this problem.

I guess that this is caused by the change in `toCallMsg`.
https://github.com/ethereum/go-ethereum/blob/v1.12.0/ethclient/ethclient.go#L606-L624
https://github.com/ethereum/go-ethereum/blob/v1.13.8/ethclient/ethclient.go#L644-L668

**Comment by bluele** [comment]
This issue is fixed at #42 @yoshidan
